liquibase-couchbase/src/test/java/liquibase/ext/couchbase/changelog/CouchbaseHistoryServiceTest.java

```diff
@@ -0,0 +1,436 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@MockitoSettings(strictness = Strictness.LENIENT)
+class CouchbaseHistoryServiceTest {
+
+    private final ChangeLogOperator changeLogOperator = mock(ChangeLogOperator.class);
+    private final ServiceProvider serviceProvider = mock(ServiceProvider.class);
+    private final CouchbaseLiquibaseDatabase database = mock(CouchbaseLiquibaseDatabase.class);
+    private CouchbaseHistoryService couchbaseHistoryService;
+
+    @BeforeEach
+    void setUp() {
+        couchbaseHistoryService = new CouchbaseHistoryService();
+        couchbaseHistoryService.setDatabase(database);
+        couchbaseHistoryService.setChangeLogOperator(changeLogOperator);
+        couchbaseHistoryService.setServiceProvider(serviceProvider);
+
+        couchbaseHistoryService.setRanChangeSetList(null);
+        couchbaseHistoryService.setLastChangeLogOrder(1);
+        couchbaseHistoryService.setInitialized(false);
+    }
+
+    @Test
+    void Should_return_priority() {
+        assertEquals(PRIORITY_SPECIALIZED, couchbaseHistoryService.getPriority());
+    }
+
+    @Test
+    void Should_init_and_create_collection_when_not_initialized_and_collection_not_created() {
+        CouchbaseConnection connection = mock(CouchbaseConnection.class);
+        Cluster cluster = mock(Cluster.class);
+        BucketManager bucketManager = mock(BucketManager.class);
+        Bucket bucket = mock(Bucket.class);
+        CollectionManager collectionManager = mock(CollectionManager.class);
+        ScopeSpec scopeSpec = mock(ScopeSpec.class);
+        List<ScopeSpec> scopeSpecs = singletonList(scopeSpec);
+        CollectionSpec collectionSpec = mock(CollectionSpec.class);
+        Set<CollectionSpec> collectionSpecs = new HashSet<>();
+        collectionSpecs.add(collectionSpec);
+
+        when(database.getConnection()).thenReturn(connection);
+        when(connection.getCluster()).thenReturn(cluster);
+        when(cluster.buckets()).thenReturn(bucketManager);
+        when(bucketManager.getBucket(SERVICE_BUCKET_NAME)).thenReturn(mock(BucketSettings.class));
+        when(cluster.bucket(SERVICE_BUCKET_NAME)).thenReturn(bucket);
+
+        when(bucket.collections()).thenReturn(collectionManager);
+        when(collectionManager.getAllScopes()).thenReturn(scopeSpecs);
+        when(scopeSpec.collections()).thenReturn(collectionSpecs);
+        when(collectionSpec.scopeName()).thenReturn(DEFAULT_SERVICE_SCOPE);
+        when(collectionSpec.name()).thenReturn("another collection");
```

Comment:
do you really need to mock all of this? 
Have you tried `mock(..class, settings().deafaultAnswers(RETURNS_DEEP_STUBS)` it may helps

Reply:
For now I need it. But I will try what you propose

Reply:
Yes it worked. I reduced a little bit mocks. Take a look.